src/main/scala/it/pagopa/interop/apigateway/api/impl/AuthApiServiceImpl.scala

```diff
@@ -0,0 +1,144 @@
+package it.pagopa.interop.apigateway.api.impl
+
+import akka.http.scaladsl.marshalling.ToEntityMarshaller
+import akka.http.scaladsl.model.StatusCodes
+import akka.http.scaladsl.server.Directives.{complete, onComplete}
+import akka.http.scaladsl.server.Route
+import cats.data.Validated.{Invalid, Valid}
+import cats.data.{NonEmptyList, Validated}
+import cats.implicits._
+import com.typesafe.scalalogging.{Logger, LoggerTakingImplicit}
+import it.pagopa.interop.apigateway.api.AuthApiService
+import it.pagopa.interop.apigateway.common.ApplicationConfiguration
+import it.pagopa.interop.apigateway.error.GatewayErrors._
+import it.pagopa.interop.apigateway.model.TokenType.Bearer
+import it.pagopa.interop.apigateway.model.{ClientCredentialsResponse, Problem}
+import it.pagopa.interop.apigateway.service.{AuthorizationManagementInvoker, AuthorizationManagementService}
+import it.pagopa.interop.authorizationmanagement.client.model.{
+  Client,
+  ClientComponentState,
+  ClientKind,
+  ClientStatesChain
+}
+import it.pagopa.pdnd.interop.commons.jwt.model.{ClientAssertionChecker, RSA, ValidClientAssertionRequest}
+import it.pagopa.pdnd.interop.commons.jwt.service.{ClientAssertionValidator, PDNDTokenGenerator}
+import it.pagopa.pdnd.interop.commons.jwt.{JWTConfiguration, JWTInternalTokenConfig}
+import it.pagopa.pdnd.interop.commons.logging.{CanLogContextFields, ContextFieldsToLog}
+import it.pagopa.pdnd.interop.commons.utils.TypeConversions._
+import it.pagopa.pdnd.interop.commons.utils.errors.ComponentError
+import org.slf4j.LoggerFactory
+
+import java.util.UUID
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.{Failure, Success, Try}
+
+final case class AuthApiServiceImpl(
+  authorizationManagementService: AuthorizationManagementService,
+  jwtValidator: ClientAssertionValidator,
+  pdndTokenGenerator: PDNDTokenGenerator
+)(implicit ec: ExecutionContext)
+    extends AuthApiService {
+
+  val logger: LoggerTakingImplicit[ContextFieldsToLog] =
+    Logger.takingImplicit[ContextFieldsToLog](LoggerFactory.getLogger(this.getClass))
+
+  lazy val jwtConfig: JWTInternalTokenConfig = JWTConfiguration.jwtInternalTokenConfig
+
+  override def createToken(
+    clientAssertion: String,
+    clientAssertionType: String,
+    grantType: String,
+    clientId: Option[String]
+  )(implicit
+    contexts: Seq[(String, String)],
+    toEntityMarshallerClientCredentialsResponse: ToEntityMarshaller[ClientCredentialsResponse],
+    toEntityMarshallerProblem: ToEntityMarshaller[Problem]
+  ): Route = {
+    val tokenAndCheckerF: Try[(String, ClientAssertionChecker)] = for {
```

Comment:
Why the final 'F'? Is something related to Try?
Maybe this approach is useful when you want to do something like 
`tokenAndChecker <- tokenAndCheckerF`
but in this case, you are doing a tuple unpacking.

It's a minor problem.